src/trip/components/tripsModal.jsx

```diff
@@ -1,39 +1,36 @@
-import React, { useEffect, useState } from 'react'
-
+import React, { useEffect, useState, useContext } from 'react'
 import { Button, Icon, Modal, Form, Input, Dropdown, Message } from 'semantic-ui-react'
+
+import { getCredentials } from '../../login/store'
+import { TripContext } from '../TripContext'
 import dateFormat from '../../utils/dateFormat'
 
 const TripsModal = ({ mode, initialData, disabled, floated, primary, isBack }) => {
-    
-    const DATA = [
-        { value: "Baptiste", text: "Baptiste" },
-        { value: "Romain", text: "Romain" },
-        { value: "Othmane", text: "Othmane" },
-        { value: "Julie", text: "Julie" },
-        { value: "Hugo", text: "Hugo" },
-        { value: "Patrick", text: "Patrick" },
-        { value: "Léa", text: "Léa" },
-        { value: "Tanguy", text: "Tanguy" },
-        { value: "Momo", text: "Momo" }
-    ]
+    const { users, createTrip, updateTrip, getUserByUsername } = useContext(TripContext)
 
+    const [modalOpen, setModalOpen] = useState(undefined)
```

Comment:
It's because i also open/close it from the outside of the component, so if i set modalOpen to false from the inside, i will never be able to open it again from the outside.

Reply:
So it's something specific to useState that if it is set to something, you can't set if from the outside, but it works if it's undefined ? sounds extremely weird

Reply:
It was not specific to useState, i finally got it right by adding an onClick event to the Modal Button, it was confusing for me that the Modals Button was inside a prop, that's why i missed it, my bad.

done.

Reply:
Okay, great ! but I guess you want to also initialize it with `false` (it's still `undefined`)

Reply:
Done